bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -49,32 +48,22 @@ type InitLocker interface {
 	Unlock(ctx context.Context, cluster *clusterv1.Cluster) bool
 }
 
-// SecretsClientFactory define behaviour for creating a secrets client
-type SecretsClientFactory interface {
-	// NewSecretsClient returns a new client supporting SecretInterface
-	NewSecretsClient(client.Client, *clusterv1.Cluster) (typedcorev1.SecretInterface, error)
-}
-
 // +kubebuilder:rbac:groups=bootstrap.cluster.x-k8s.io,resources=kubeadmconfigs;kubeadmconfigs/status,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=cluster.x-k8s.io,resources=clusters;clusters/status;machines;machines/status,verbs=get;list;watch
 // +kubebuilder:rbac:groups="",resources=secrets;events;configmaps,verbs=get;list;watch;create;update;patch;delete
 
 // KubeadmConfigReconciler reconciles a KubeadmConfig object
 type KubeadmConfigReconciler struct {
 	client.Client
```

Comment:
While you're in here, would you mind giving this field a name, instead of keeping the embedding?

Reply:
updated!